databases/experiment_database.py

```diff
@@ -1,445 +1,512 @@
-            animals = self._c.fetchall()
-            ans =[]
-            for animal in animals:
-                ans.append(str(animal[0]))
-            animals_by_group[str(group[1])] = ans
-
-        return animals_by_group
-
-    def get_cages_by_group(self):
-        '''Returns a dict where the key is group_id and
-        the value is a list of cages in that group.'''
-        self._c.execute("SELECT group_id, name FROM groups", ())
-        groups = self._c.fetchall()
-
-        cages_by_group = {}
-
-        for group in groups:
-            self._c.execute("SELECT cage_id FROM cages WHERE group_id=?", (group[0],))
-            cages = self._c.fetchall()
-
-            ans = []
-            for cage in cages:
-                ans.append(str(cage[0]))
-            cages_by_group[str(group[1])] = ans
-
-        return cages_by_group
-
-
-    def get_animals(self):
-        '''Returns a list of tuples representing each animal in experiment.'''
-        self._c.execute("SELECT animal_id, group_id, cage_id, weight, active FROM animals")
-        return self._c.fetchall()
-
-    def get_animal_id(self, rfid):
-        '''Returns the id of the animal with the passed RFID.'''
-        self._c.execute("SELECT animal_id FROM animal_rfid WHERE rfid=?", (rfid,))
-        return self._c.fetchone()[0]
-
-    def get_animal_rfid(self, animal_id):
-        '''Returns the rfid number of the passed animal'''
-        self._c.execute("SELECT rfid FROM animal_rfid WHERE animal_id=?", (int(animal_id),))
-        return self._c.fetchone()[0]
-
-    def get_animals_rfid(self):
-        '''Returns a list of all animal rfids.'''
-        self._c.execute("SELECT rfid FROM animal_rfid")
-        return self._c.fetchall()
-
-    def get_all_experiment_info(self):
-        '''Prints all experiment info to terminal.'''
-        self._c.execute("SELECT name, species, uses_rfid, num_animals, num_groups, cage_max FROM experiment")
-        print('Experiment: name, species, uses_rfid, num_animals, num_groups, cage_max')
-        print(self._c.fetchall())
-        self._c.execute("SELECT name, num_animals, full FROM groups")
-        print('Groups: name, num_animals, full')
-        print(self._c.fetchall())
-        self._c.execute("SELECT cage_id, group_id, num_animals, full FROM cages")
-        print('cages: cage_id, group_id, num_animals, full')
-        print(self._c.fetchall())
-        self._c.execute("SELECT item, auto FROM measurement_items")
-        print('Measurement Items: item, auto')
-        print(self._c.fetchall())
-
-    def close(self):
-        '''Closes database file.'''
-        self._conn.close()
+'''SQLite Database module for Mouser.'''
+import sqlite3
+import os
+import csv
+from customtkinter import CTk, CTkLabel, CTkButton
+
+class ExperimentDatabase:
+    '''SQLite Database Object for Experiments.'''
+    def __init__(self, file=":memory:"):  #call with file name as argument or no args to use memory
+        self.db_file = file

```

Comment:
This line is the only thing in the first 400 lines that I actually changed. But it pushed everything else down in line numbers